src/app/CommandHandler.cpp

```diff
@@ -92,40 +99,20 @@ CHIP_ERROR CommandHandler::AllocateBuffer()
     return CHIP_NO_ERROR;
 }
 
-void CommandHandler::OnInvokeCommandRequest(Messaging::ExchangeContext * ec, const PayloadHeader & payloadHeader,
-                                            System::PacketBufferHandle && payload, bool isTimedInvoke)
+Status CommandHandler::OnInvokeCommandRequest(CommandResponderInterface & commandResponder, System::PacketBufferHandle && payload,
+                                              bool isTimedInvoke)
 {
-    System::PacketBufferHandle response;
-    Status status = Status::Failure;
-    VerifyOrDieWithMsg(ec != nullptr, DataManagement, "Incoming exchange context should not be null");
     VerifyOrDieWithMsg(mState == State::Idle, DataManagement, "state should be Idle");
 
-    // NOTE: we already know this is an InvokeCommand Request message because we explicitly registered with the
-    // Exchange Manager for unsolicited InvokeCommand Requests.
-    mResponseSender.SetExchangeContext(ec);
+    SetCommandResponder(&commandResponder);
```

Comment:
I was trying to look into the lifetiume of the commandResponder ... we set it oninvoke or test invoke, however it seems never reset. It seems potentially desirable to reset it back to invalid once the invoke processing is completed, so it is clear when it would be safe to be released by whoever uses the command handler.

Reply:
I think setting to null on close is better - it would show that the lifetime was considered. It may also help in case we have bugs that access data past close.